datahub-web-react/src/app/home/HomePageHeader.tsx

```diff
@@ -57,8 +110,37 @@ export const HomePageHeader = () => {
         });
     };
 
+    // fetch some results from each entity to display search suggestions
+    const allSearchResultsByType = useGetAllEntitySearchResults({
+        query: '*',
+        start: 0,
+        count: 20,
+        filters: [],
+    });
+
+    const suggestionsLoading = Object.keys(allSearchResultsByType).some((type) => {
+        return allSearchResultsByType[type].loading;
+    });
+
+    const suggestionsToShow = useMemo(() => {
+        let result: string[] = [];
+        if (!suggestionsLoading) {
+            [EntityType.Dashboard, EntityType.Chart, EntityType.Dataset].forEach((type) => {
+                const suggestionsToShowForEntity = getSuggestionFieldsFromResult(
+                    allSearchResultsByType[type]?.data,
+                ).sort(sortRandom);
+                const suggestionToAddToFront = suggestionsToShowForEntity?.pop();
+                result = [...result, ...suggestionsToShowForEntity];
+                if (suggestionToAddToFront) {
```

Comment:
What is this logic doing?

Reply:
This interleaves the results of different types of queries so we have a variety of entity types suggested

Reply:
Got it